### PR TITLE
🌱 Add github action to trigger container image build

### PR DESCRIPTION
**Pull request by mquhuy**
This PR adds an action to trigger capm3 container build whenever a PR is merged in this repo.

### CODE REVIEW

**Review by kashifest — COMMENTED** [review]
can we hold this and others and test if the IPAM PR https://github.com/metal3-io/ip-address-manager/pull/386 is working as expected first?

**Comment by mquhuy** [comment]
> can we hold this and others and test if the IPAM PR [metal3-io/ip-address-manager#386](https://github.com/metal3-io/ip-address-manager/pull/386) is working as expected first?

Sure. For now we can test out by merging one of them

File: .github/workflows/build-images-action.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: build-images-action
+on:
+  push:
+    branches:
+      - main
+      - 'release-*'
```
[the remark this answers is not in the document]
Comment (mquhuy) [reply .github/workflows/build-images-action.yml:6]: tbh I just copied the practices from GH actions doc. My assumption is that if the string has special characters, like dash ('-'), single quotation marks are expected.

File: .github/workflows/build-images-action.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      - name: build capm3 image
+        uses: toptal/jenkins-job-trigger-action@137fff703dd260b52b53d3ba1960396415abc568 # 1.0.2
+        with:
+          jenkins_url: "https://jenkins.nordix.org/"
+          jenkins_user: "metal3.bot@gmail.com"
+          jenkins_token: ${{ secrets.JENKINS_TOKEN }}
+          job_name: "metal3_cluster-api-provider-metal3_container_image_building"
+          job_params: |
+            {
+              "BUILD_CONTAINER_IMAGE_NAME": "baremetal-operator",
```
[the remark this answers is not in the document]
Comment (mquhuy) [reply .github/workflows/build-images-action.yml:27]: Oops

File: .github/workflows/build-images-action.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mquhuy) [reply .github/workflows/build-images-action.yml:6]: Hmm. Okay then

**Review by smoshiur1237 — COMMENTED** [review]
/lgtm

**Review by tuminoid — COMMENTED** [review]
/lgtm

**Comment by tuminoid** [comment]
/override test-centos-e2e-integration-main test-ubuntu-integration-main

**Comment by tuminoid** [comment]
I've disabled the builld trigger in Quay for CAPM3. All builds should flow thru this workflow + Jenkins now.

**Pull request merged**